From 277af3dffdfe44f8409d77d7b3f4444a12f8bb2c Mon Sep 17 00:00:00 2001
From: Rohit Visavalia <rohit.visavalia@xilinx.com>
Date: Tue, 6 Oct 2020 09:30:31 -0700
Subject: [PATCH 1702/1851] staging: xlnxsync: Remove channel interrupt mask
 handling at enable/disable

commit 933bf06c2ce4a5ce76de47b075bd37712e5cfbb5 from
https://github.com/Xilinx/linux-xlnx.git

By default all interrupts should be enabled, so remove the masking logic
at channel enable and the corresponding unmasking logic at channel disable.

Signed-off-by: Rohit Visavalia <rohit.visavalia@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/staging/xlnxsync/xlnxsync.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/drivers/staging/xlnxsync/xlnxsync.c b/drivers/staging/xlnxsync/xlnxsync.c
index b6dc396242a5..72aaf6cc7bc0 100644
--- a/drivers/staging/xlnxsync/xlnxsync.c
+++ b/drivers/staging/xlnxsync/xlnxsync.c
@@ -584,8 +584,6 @@ static int xlnxsync_chan_enable(struct xlnxsync_channel *channel, bool enable)
 
 	if (enable) {
 		dev_dbg(dev->dev, "Enabling %d channel\n", channel->id);
-		xlnxsync_set(dev, channel->id, XLNXSYNC_IER_REG,
-			     XLNXSYNC_IER_ALL_MASK);
 		xlnxsync_set(dev, channel->id, XLNXSYNC_CTRL_REG,
 			     XLNXSYNC_CTRL_ENABLE_MASK |
 			     XLNXSYNC_CTRL_INTR_EN_MASK);
@@ -595,8 +593,6 @@ static int xlnxsync_chan_enable(struct xlnxsync_channel *channel, bool enable)
 		xlnxsync_clr(dev, channel->id, XLNXSYNC_CTRL_REG,
 			     XLNXSYNC_CTRL_ENABLE_MASK |
 			     XLNXSYNC_CTRL_INTR_EN_MASK);
-		xlnxsync_clr(dev, channel->id, XLNXSYNC_IER_REG,
-			     XLNXSYNC_IER_ALL_MASK);
 		channel->prod_sync_err = false;
 		channel->prod_wdg_err = false;
 		channel->cons_sync_err = false;
@@ -945,9 +941,6 @@ static int xlnxsync_release(struct inode *iptr, struct file *fptr)
 		xlnxsync_clr(dev, channel->id, XLNXSYNC_CTRL_REG,
 			     XLNXSYNC_CTRL_ENABLE_MASK |
 			     XLNXSYNC_CTRL_INTR_EN_MASK);
-		xlnxsync_clr(dev, channel->id, XLNXSYNC_IER_REG,
-			     XLNXSYNC_IER_ALL_MASK);
-
 	}
 
 	if (mutex_lock_interruptible(&dev->sync_mutex))
-- 
2.31.1

